### PR TITLE
ShedulerService API #8597

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/SchedulerName.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.enonic.xp.annotation.PublicApi;
+
+@PublicApi
+public final class SchedulerName
+{
+    private final String value;
+
+    private SchedulerName( final String value )
+    {
+        validate();
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/SchedulerName.java:16]: Move validation out of private constructors

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/CreateScheduledJobParams.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    {
+        return name;
+    }
+
+    public String getDescription()
+    {
+        return description;
+    }
+
+    public String getFrequency()
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/CreateScheduledJobParams.java:61]: Let's consider  a custom type

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/CreateScheduledJobParams.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+        private PropertyTree payload;
+
+        private PrincipalKey user;
+
+        private PrincipalKey author;
+
+        public Builder()
+        {
+            super();
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/CreateScheduledJobParams.java:118]: Useless constrctor

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/EditableScheduledJob.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.TimeZone;
+
+import com.enonic.xp.annotation.PublicApi;
+import com.enonic.xp.data.PropertyTree;
+import com.enonic.xp.page.DescriptorKey;
+import com.enonic.xp.security.PrincipalKey;
+
+@PublicApi
+public class EditableScheduledJob
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/EditableScheduledJob.java:12]: final

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/ScheduledJobEditor.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.enonic.xp.scheduler;
+
+import com.enonic.xp.annotation.PublicApi;
+
+@PublicApi
+public interface ScheduledJobEditor
+{
+    void edit( final EditableScheduledJob edit );
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/ScheduledJobEditor.java:8]:
```suggestion
    void edit( EditableScheduledJob edit );
```

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/core/impl/scheduler/SchedulerServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.List;
+
+import com.enonic.xp.scheduler.CreateScheduledJobParams;
+import com.enonic.xp.scheduler.ModifyScheduledJobParams;
+import com.enonic.xp.scheduler.ScheduledJob;
+import com.enonic.xp.scheduler.SchedulerName;
+import com.enonic.xp.scheduler.SchedulerService;
+
+public final class SchedulerServiceImpl
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/core/impl/scheduler/SchedulerServiceImpl.java:11]:
```suggestion
public class SchedulerServiceImpl
```

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/SchedulerService.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.enonic.xp.scheduler;
+
+import java.util.List;
+
+public interface SchedulerService
+{
+    ScheduledJob create( final CreateScheduledJobParams params );
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/SchedulerService.java:7]:
```suggestion
    ScheduledJob create( CreateScheduledJobParams params );
```

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/core/impl/scheduler/SchedulerServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    implements SchedulerService
+{
+    @Override
+    public ScheduledJob create( final CreateScheduledJobParams params )
+    {
+        return null;
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/core/impl/scheduler/SchedulerServiceImpl.java:17]: throw not implemented

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/ScheduledJobEditor.java (hunk 5 above)
Comment (vbradnitski) [reply modules/core/core-api/src/main/java/com/enonic/xp/scheduler/ScheduledJobEditor.java:8]: haven't got this one

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/core/impl/scheduler/SchedulerServiceImpl.java (hunk 6 above)
Comment (vbradnitski) [reply modules/core/core-scheduler/src/main/java/com/enonic/xp/core/impl/scheduler/SchedulerServiceImpl.java:11]: and this one

File: modules/core/core-api/src/main/java/com/enonic/xp/scheduler/Frequency.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
+package com.enonic.xp.scheduler;
+
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+
+import com.enonic.xp.annotation.PublicApi;
+
+@PublicApi
+public final class Frequency
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/scheduler/Frequency.java:10]: I will consider it to be interface

**Review by rymsha — APPROVED** [review]

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**